### ISSUE TITLE
Remove duplicated frontmatter keys from localized content

### ISSUE THREAD

**Issue by queengooborg** [mdn/translated-content#7412]
As pointed out by @yin1999, frontmatter metadata (ex. tags, title) is [combined with the metadata from the original en-US docs](https://github.com/mdn/yari/blob/5d8d88976164b6f653c3f2de688f3cd76f8537bb/kumascript/index.js#L56-L69).  As such, keys such as tags, browser-compat, and spec-urls is redundant to copy to the localized content.  These duplicated keys should be removed from the localized docs for easier maintenance overall.  We should only keep: `title`, `short_title`, `slug`, ~~`original_slug`~~ and `l10n.*`.

**Edit; July 18, 2023**
A [configuration file](https://github.com/mdn/translated-content/blob/main/front-matter-config.json) is now present in this repository, which will let us utilize the frontmatter linter in upstream content!  To use the linter to fix the frontmatter, perform the following:

```sh
cd /path/to/mdn/content
yarn fix:fm --configFile /path/to/mdn/translated-content/front-matter-config.json --files /path/to/mdn/translated-content/files/<locale>
```

**Comment by SphinxKnight** [comment]
@queengooborg is there any "deadline" or something blocked by this or can we go progressively during "new" edits on pages?

Note to self: double-check `HTMLRef` (and maybe JS or CSS sidebar as I have a doubt about them not using localized pages' tags)

**Comment by queengooborg** [comment]
This can be done progressively and isn't blocking anything, so it can be done at our own leisure!

**Comment by queengooborg** [comment]
In another issue (#5618), @yin1999 pointed out that all frontmatter keys are merged with the en-US metadata.  I've broadened the scope of this issue to include _all_ redundant metadata!

**Comment by SphinxKnight** [comment]
Without testing, I'd go for removing `translation_of`, this key might be useful for dealing with `orphaned`/`conflicting` pages but does not contain any other info than the slug (in a yari world).

**Comment by yin1999** [comment]
Done for l10n-zh

**Comment by YujiSoftware** [comment]
I made a script to batch delete.
https://gist.github.com/YujiSoftware/bfa7aa0957d276c679f52d254baea6b1

Remove anything but the meta below.
* title
* slug
* original_slug
* l10n.*
* i10n.*

**Comment by hochan222** [comment]
Done for l10n-ko (Thank you @yin1999)

**Comment by hochan222** [comment]
> We should only keep: title, slug, original_slug and l10n.*.

@queengooborg  Hello :)

If you look at [yari /content/document.ts L106-L114](https://github.com/mdn/yari/blob/fb11ce6b175a1c4cc9acbf063966be0b29cbe83e/content/document.ts#L106-L114), you can see that only "title" and "slug" are included in requiredFrontMatterKeys. "original_slug" is included in optionalFrontMatterKeys. Can I remove "original_slug" from translated-content as it is no longer required?

I wonder why you wrote earlier that if you have to keep it mandatory, it should be included.

(cc. @yin1999 )

**Comment by queengooborg** [comment]
I was going to say that we should leave it in, but looking through the code, it appears that `original_slug` isn't used anywhere, but rather it's only set during translated content sync.  I don't understand why a redirect isn't being made instead...

Given it's not used, I think it's safe to remove.  👍

**Comment by yin1999** [comment]
> Can I remove "original_slug" from translated-content as it is no longer required?

Yes, we could remove this from documents, as this is used for indicate where the document is moved from (used by [sync-translated-content.yml](https://github.com/mdn/translated-content/blob/9726fc021d2a7f9ab385ad5dbe575254bb39d6a4/.github/workflows/sync-translated-content.yml)).

> I wonder why you wrote earlier that if you have to keep it mandatory, it should be included.

I'm not sure if you mean "we need to add `short-title` in to the list. If so, that's because when we use `yarn content move` to move document, the frontmatter key's order may be changed, as that tool uses the list to recreate the front-matter for the moved document.

**Comment by hochan222** [comment]
Thank you all for your answers.

I'll correct the locale guide. => We should only keep: `title`, `slug` and `l10n.*`.

---

> I'm not sure if you mean "we need to add short-title in to the list".

@yin1999 

It wasn't meant for `short-title`.(`l10n.*` was also not included in requiredFrontMatterKeys) 
I meant the following sentence mentioned at the top of this article. It was just for `original_slug`. 

**"We should only keep: title, slug, original_slug and l10n.*."**

The reason for mentioning it is that you were mentioned at the top, so you can answer well about it.

---

I got enough answers. Thank you. Have a nice weekend :)

**Comment by queengooborg** [comment]
After speaking with the rest of the MDN team, they pointed out that `short_title` is also localizable so that should be kept in translated content too!

**Comment by Lou8is** [comment]
I have started applying this to https://github.com/mdn/translated-content/labels/l10n-fr however the PR would be sort of massive. 
My branch is ready (see https://github.com/mdn/translated-content/compare/main...Lou8is:translated-content:%237412-cleans-metadata), with 1 commit per sub-folder.

Should I do 1 PR per commit or something? I'll wait for input before submitting. Thanks!

**Comment by queengooborg** [comment]
> I have started applying this to [ l10n-fr](https://github.com/mdn/translated-content/labels/l10n-fr) Issues related to French content. however the PR would be sort of massive. My branch is ready (see [main...Lou8is:translated-content:#7412-cleans-metadata](https://github.com/mdn/translated-content/compare/main...Lou8is:translated-content:%237412-cleans-metadata)), with 1 commit per sub-folder.
> 
> Should I do 1 PR per commit or something? I'll wait for input before submitting. Thanks!

Given the size of your branch and the number of files changed, I would say yes, dividing it into multiple PRs based upon subfolders would be great!

**Comment by Lou8is** [comment]
> > I have started applying this to [ l10n-fr](https://github.com/mdn/translated-content/labels/l10n-fr) Issues related to French content. however the PR would be sort of massive. My branch is ready (see [main...Lou8is:translated-content:#7412-cleans-metadata](https://github.com/mdn/translated-content/compare/main...Lou8is:translated-content:%237412-cleans-metadata)), with 1 commit per sub-folder.
> > Should I do 1 PR per commit or something? I'll wait for input before submitting. Thanks!
> 
> Given the size of your branch and the number of files changed, I would say yes, dividing it into multiple PRs based upon subfolders would be great!

Thanks for the quick answer, I'll do just that.
